705.py

```diff
@@ -0,0 +1,50 @@
+# N.B.
+# We should go back visit linkedlist and BST solutions.
+# We should do time complexity analysis for each HashSet function.
+
+class MyHashSet:
+
+    def __init__(self):
+        """
+        Initialize your data structure here.
+        """
+        self.SPACE_SIZE = 9999
+        self.space = [[] for _ in range(self.SPACE_SIZE)]
+
+    def add(self, key: int) -> None:
+        hashcode = self.computeHash(key)
+
+        for k in self.space[hashcode]:
```

Comment:
This `for k in self.space[loc]:` operation has a linear time complexity. It takes `O(N)` time to find `k` from `self.space[loc]`, assuming there are `N` elements in your linked list `self.space[loc]`. If somehow all the elements collide into a specific slot in `self.space`, the overall performance of your current hash set design degrades to a simple linked list's performance.

This same issue exists in your `remove()` and `contains()` implementations as well.